src/main/java/frc/robot/commands/StateMachine.java

```diff
@@ -107,9 +137,17 @@ private boolean isScoreState(RobotState state) {
 
     private void publishStates() {
         currentStatePub.set(state.toString());
+        elevatorStatePub.set(state.elevatorState.toString());
         intakeStatePub.set(state.intakeState.toString());
         manipulatorStatePub.set(state.manipulatorState.toString());
-        elevatorStatePub.set(state.elevatorState.toString());
     }
 
+    public void setTargetScoreState(RobotState targetState) {
+        if (isScoreState(targetState)) {
+            this.targetScoreState = targetState;
+            System.out.println("Target score state set to: " + targetState);
+        }
+    }
+
+
 }
```

Comment:
instead of making the state machine a command, make it a subsystem so if you want to continually publish the states to network tables you can do so in the periodic method